File: sams-backend/src/main/java/com/sams/entity/Store.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_store", schema = "inventory")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Store {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "store_id")
    private Long storeId;

    @Column(name = "store_name")
    private String storeName;

    @Column(name = "store_code")
    private String storeCode;

    @Column(name = "store_description")
    private String storeDescription;

    @Column(name = "incharge_name")
    private String inchargeName;

    @Column(name = "incharge_id")
    private String inchargeId;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt_disp")
    private String createdDtDisp;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt_disp")
    private String updatedDtDisp;

    @PrePersist
    protected void onCreate() {
       // Auto-generated
    }

    @PreUpdate
    protected void onUpdate() {
       // Auto-generated
    }
}

File: SAMSUI/src/app/Model/master/store.ts
import { BaseModel } from '../base/baseModel';

export class StoreModel extends BaseModel{

    public storeId?: number;
    public storeName?: string;
    public storeCode?: string;
    public storeDescription?: string;
    public inchargeName?: string;
    public inchargeId?: string;
    public pageNumber ?: number;
    public recordsPerPage ?: number;
    public createdBy ?:string;
    public createdDtDisp ?: string;
    public updatedBy ?: string;
    public updatedDtDisp ?: string;


}